### PR TITLE
✨Add custom onPicked callback when asset entity is generated

### PR DESCRIPTION
**Pull request by yujune**
### Context
Currently, when a picture or video is taken using the camera, the AssetEntity is retrieved via:
```
final assetEntity = await pickFromCamera(context);
```
This is achieved by calling `Navigator.of(context).pop(entity);`, which returns the `AssetEntity` to the previous screen.

However, this approach has a limitation when the `CameraPicker` widget is used in a scenario where it doesn't trigger a pop event, such as when the widget is part of a tab or a persistent page in a navigation stack. In these cases, there is no straightforward way to retrieve the captured `AssetEntity`.

### Changes
- Introduced a custom `onPicked` callback that directly provides the generated AssetEntity without relying on the `Navigator.of(context).pop(entity);`.

### CODE REVIEW

File: lib/src/states/camera_picker_state.dart [hunk 1]
```diff
@@ -926,7 +926,12 @@ class CameraPickerState extends State<CameraPicker>
         viewType: CameraPickerViewType.image,
       );
       if (entity != null) {
-        Navigator.of(context).pop(entity);
+        if (pickerConfig.onPicked case final onPicked?) {
+          await controller.resumePreview();
+          onPicked(entity);
```
Comment (AlexV525) [inline lib/src/states/camera_picker_state.dart:931]: `resumePreview` doesn't seem to be enough at this place.

**Review by AlexV525 — COMMENTED** [review]
The idea is good, I'd prefer `onPickConfirmed` for better readability.

File: lib/src/constants/config.dart [hunk 2]
```diff
@@ -165,4 +167,8 @@ final class CameraPickerConfig {
   /// The callback when the recording is not met the minimum recording duration.
   /// 录制时长未达到最小时长时的回调方法。
   final VoidCallback? onMinimumRecordDurationNotMet;
+
+  /// The callback when picture is taken or video is confirmed.
+  /// 拍照或录像确认时的回调方法。
+  final Function(AssetEntity)? onPicked;
```
Comment (AlexV525) [inline lib/src/constants/config.dart:173]: The return type needs to be identified.

File: lib/src/states/camera_picker_state.dart (hunk 1 above)
Comment (yujune) [reply lib/src/states/camera_picker_state.dart]: done updated to `onPickConfirmed`.

File: lib/src/constants/config.dart (hunk 2 above)
Comment (yujune) [reply lib/src/constants/config.dart]: done added `void` as return type.

File: lib/src/states/camera_picker_state.dart (hunk 1 above)
Comment (yujune) [reply lib/src/states/camera_picker_state.dart]: any suggestion for this?

Comment (AlexV525) [reply lib/src/states/camera_picker_state.dart]: There are more steps after the return, so maybe we can put the pop step together with the return.

Comment (yujune) [reply lib/src/states/camera_picker_state.dart]: done putting return together with pop step.

File: lib/src/constants/config.dart [hunk 3]
```diff
@@ -165,4 +167,8 @@ final class CameraPickerConfig {
   /// The callback when the recording is not met the minimum recording duration.
   /// 录制时长未达到最小时长时的回调方法。
   final VoidCallback? onMinimumRecordDurationNotMet;
+
+  /// The callback when picture is taken or video is confirmed.
```
Comment (AlexV525) [inline lib/src/constants/config.dart:171]:
```suggestion
  /// The callback when the picture or the video is confirmed as picked.
```

**Review by AlexV525 — APPROVED** [review]
Thanks for following up all suggestions!

**Comment by AlexV525** [comment]
@all-contributors Add @yujune for code

**Pull request merged**